source/advanced-topics.adoc

```diff
@@ -1143,3 +1143,136 @@ The `%ifos` conditional is used to control processing based on the operating sys
+
+ifdef::rhel[]
+
+[source,specfile]
+----
+Name:           python-setuptools <1>
+Version:        0.9.8
+Release:        1%{?dist}
+Summary:        Easily build and distribute Python packages
+License:        Python or ZPLv2.0
+URL:            https://pypi.org/project/setuptools/
+Source0:        https://files.pythonhosted.org/packages/source/s/setuptools/setuptools-%{version}.zip
+
+BuildArch:      noarch
+
+BuildRequires:  python2-devel <2>
+
+BuildRequires:  python-backports-ssl_match_hostname <3>
+Requires:       python-backports-ssl_match_hostname
+
+%description
+Setuptools is a collection of enhancements to the Python distutils that allow
+you to more easily build and distribute Python packages, especially ones that
+have dependencies on other packages.
+
+This package contains the runtime components of setuptools, necessary to
+execute the software that requires pkg_resources.py.
+
+%prep
+%setup -q -n setuptools-%{version}
+
+%build
+CFLAGS="$RPM_OPT_FLAGS" %{__python} setup.py build <4>
+
+%install
+%{__python} setup.py install --skip-build --root %{buildroot} <5>
+
+%check
+LANG=en_US.utf8 %{__python} setup.py test <6>
+
+%files
+%doc *.txt docs
+%{python_sitelib}/*
+%{_bindir}/easy_install
+
+%changelog
+...
+----
+
+<1> The name of this package in PyPI is setuptools, in RPM this package has specification "python" in its name.
+<2> BuildRequires specifies what packages are required to to build this package.
+<3> BuildRequires specifies which Python module package python-setuptools requires.
+<4> Without this expression the package might not be possible to build if the package has a C extension. `%{__python} setup.py build` Is similar to make build. Allows to build Python packages.
+<5> The setup.py install is similar to make install. On this line, setup.py install would suffice because it builds and installs the package, but for keeping an RPM standard the package is build with setup.py build and installed with the --skip-build option.
+<6> Tests vary depending on a particular project. However in RHEL locale has to be specified on LANG=en_US.utf8.
+
+endif::rhel[]
+
+ifdef::community[]
+
+// Next: Use spec file to desribe typical python RPM package in Fedora. https://src.fedoraproject.org/rpms/python-setuptools/blob/master/f/python-setuptools.spec
+
+endif::community[]
+
+[id="Common-macros-for-Python-RPM-packages"]
+==== Common macros for Python RPM packages
+
+ifdef::community[]
+
+All Python specific RPM macros can be found on this page link:https://fedoraproject.org/wiki/Packaging:Python#Macros[Python packaging macros].
+
+endif::community[]
+
+ifdef::rhel[]
+
+|===
+|Macro |Normal Definition |Notes
+
+|__python
+|/usr/bin/python
+|Python 2 interpreter
+
+|python_sitelib
+|/usr/lib/python.X.Y/site-packages
+|Where pure python modules are installed
```

Comment:
Could you please make a suggestion what are those modules and what does "pure" means?

Reply:
python modules https://docs.python.org/2/tutorial/modules.html

pure python - written only in Python (as opposed to [extension modules](https://docs.python.org/2/extending/extending.html) compiled from languages like C, C++ or Fortran)